src/rules/__tests__/normalizeSyntheticEvent.test.js

```diff
@@ -111,4 +111,70 @@ describe('normalizeSyntheticEvent', function () {
       }
     });
   });
+
+  it(
+    'does not override meta of first normalized event if synthetic event is ' +
+      'normalized multiple times',
+    function () {
+      var syntheticEvent = {};
+      var syntheticEventMeta1 = { $rule: { id: 'ABC123' } };
+      var syntheticEventMeta2 = { $rule: { id: 'ABC456' } };
+      var normalizedSyntheticEvent1 = normalizeSyntheticEvent(
+        syntheticEventMeta1,
+        syntheticEvent
+      );
+      var normalizedSyntheticEvent2 = normalizeSyntheticEvent(
+        syntheticEventMeta2,
+        syntheticEvent
+      );
+
+      expect(normalizedSyntheticEvent1.$rule.id).toEqual('ABC123');
+      expect(normalizedSyntheticEvent2.$rule.id).toEqual('ABC456');
+    }
+  );
+
+  it('leaves syntheticEvent objects alone that are not plain objects', function () {
+    // source: https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent
+    function CustomEventShim(event, params) {
+      params = params || {
+        bubbles: false,
+        cancelable: false,
+        detail: undefined
+      };
+      var evt = document.createEvent('CustomEvent');
+      evt.initCustomEvent(
+        event,
+        params.bubbles,
+        params.cancelable,
+        params.detail
+      );
+      return evt;
+    }
+
+    if (typeof window.CustomEvent !== 'function') {
+      CustomEventShim.prototype = window.Event.prototype;
+      window.CustomEvent = CustomEventShim;
```

Comment:
It would be best to either not set `window.CustomEvent` or clean it up after the test. If, for example, we added code to Turbine that expected `window.CustomEvent` to exist, then a test should probably fail when run on IE since `window.CustomEvent` doesn't actually exist in IE. Because you're setting `window.CustomEvent` here, though, it may mask the issue if the test that _should_ fail runs after this one.

I haven't tried it, but I think you might be able to do:

```
var CustomEvent = typeof window.CustomEvent === 'function' ? window.CustomEvent.bind(window) : CustomEventShim.bind(window);
```

...or something like that.